### PR TITLE
DR-1154 bulk driver bug

### PR DESCRIPTION
**Pull request by ddietterich**
This change set has a few small changes to accommodate interface changes from stairway.
It also teaches the DataRepoClient tolerate 502 errors that come up when a pod is killed in the middle of processing a job polling request.

### CODE REVIEW

File: src/test/java/bio/terra/integration/DataRepoClient.java [hunk 1]
```diff
@@ -87,37 +87,66 @@ public DataRepoClient() {
...
+                sleepSeconds = Math.min(2 * sleepSeconds, maxSeconds);
+
+                DataRepoResponse<JobModel> jobModelResponse = get(user, location, JobModel.class);
+                logger.info("Got response. status: " + jobModelResponse.getStatusCode()
+                    + " location: " + jobModelResponse.getLocationHeader().orElse("not present"));
+                if (jobModelResponse.getStatusCode() != notStatus) {
+                    return jobModelResponse;
+                }
             }
         } catch (InterruptedException ex) {
```
Comment (snf2ye) [inline src/test/java/bio/terra/integration/DataRepoClient.java:154]: I encountered some IOExceptions here when I was doing my Kubernetes testing. Do you think that would be worth handling here?

**Review by snf2ye — APPROVED** [review]
I like your design for the wait for response! 
One question, but looks good to me.

**Comment by ddietterich** [comment]
Analyzed the failure. Integration failed due to SAM down. Connected failed due to DR-1162.

File: src/test/java/bio/terra/integration/DataRepoClient.java (hunk 1 above)
Comment (ddietterich) [reply src/test/java/bio/terra/integration/DataRepoClient.java:154]: Sure. What do you do when you catch one? Can I copy some code??

**Review by fboulnois — COMMENTED** [review]
lgtm, can you reuse the retry mechanisms you've already developed previously for the `while` loop sections (they seem like they'd be more robust)?

**Comment by ddietterich** [comment]
Connected failed due to DR-1162. I will be very glad when that gets fixed!

File: src/test/java/bio/terra/integration/DataRepoClient.java (hunk 1 above)
Comment (snf2ye) [reply src/test/java/bio/terra/integration/DataRepoClient.java:154]: Thinking about it, this only makes sense if you want to add some sort of retry mechanism. I followed the pattern in GoogleResourceService -> enableIamPermissions.

**Review by fboulnois — APPROVED** [review]
the switch to using a terminating condition lgtm!

**Pull request merged**